sbol_utilities/component.py

```diff
@@ -572,6 +572,176 @@ def backbone(identity: str, sequence: str, dropout_location: List[int], fusion_s
+    roles_set = set(part_insert_component.roles)
+    types_set.add(sbol3.SO_DOUBLE_STRANDED)
+    types_set.add(part.types)
+    roles_set.add(tyto.SO.engineered_insert) #review
+    roles_set.add(part.roles)
+    part_insert_component.types = list(types_set)
+    part_insert_component.roles = list(roles_set)
+    upstream_fusion_site_location = sbol3.Range(sequence=part_insert_seq, start=1, end=len(fusion_site_sequences[0]))
+    downstream_fusion_site_location = sbol3.Range(sequence=part_insert_seq, start=len(part_insert_seq_str)-len(fusion_site_sequences[1]), end=len(part_insert_seq_str))
+    upstream_fusion_site_feature = sbol3.SequenceFeature(locations=[upstream_fusion_site_location], roles=[tyto.SO.restriction_enzyme_five_prime_single_strand_overhang])
+    downstream_fusion_site_feature = sbol3.SequenceFeature(locations=[downstream_fusion_site_location], roles=[tyto.SO.restriction_enzyme_three_prime_single_strand_overhang])
+    part_insert_component.features.append(upstream_fusion_site_feature)
+    part_insert_component.features.append(downstream_fusion_site_feature)
+    return part_insert_component, part_insert_seq
+
+def part_in_backbone(identity: str, part: sbol3.Component, backbone: sbol3.Component, linear:bool=False, **kwargs) -> Tuple[sbol3.Component, sbol3.Sequence]:
+    """Creates a Part in Backbone Component and its Sequence.
+
+    :param identity: The identity of the Component. The identity of Sequence is also identity with the suffix '_seq'.
+    :param part: Part to be located in the backbone as SBOL Component.
+    :param backbone: Backbone in wich the part is located as SBOL Component.
+    :param linear: Boolean than indicates if the backbone is linear, by default it is seted to Flase which means that it has a circular topology.
+    :param kwargs: Keyword arguments of any other Component attribute.
+    :return: A tuple of Component and Sequence.
+    """
+    #TODO check extreme from the part and backbone, if they match pass, if not raise error 
+    # check that backbone has a plasmid vector or child ontology term
+    if any(role==tyto.SO.plasmid_vector for role in backbone.roles): # how to check if child also?
+        pass
+    else: raise TypeError('The backbone has no valid plasmid vector or child role')
+    # get backbone sequence
+    backbone_sequence = backbone.sequences[0].lookup().elements
+    # extract backbone open sequence
+    open_backbone_sequence = backbone_sequence[backbone.features[2].locations[0].start -1 : -1] + backbone_sequence[0 : backbone.features[2].locations[0].end]
+    # make sure there are recognition sites
+    # extract part sequence
+    part_sequence = part.sequences[0].lookup().elements
+    # make new component sequence
+    part_in_backbone_seq_str = part_sequence + open_backbone_sequence
+    # part in backbone Component
+    part_in_backbone_component, part_in_backbone_seq = dna_component_with_sequence(identity, part_in_backbone_seq_str, **kwargs)
```

Comment:
I think compute_sequence works for parts with only 1 sequence, and when nothing got removed. After an inspection of the function I think it will not take into account that the sequence for `backbone` that I developed as the insert. So it will concatenate the sequence of the insert to the sequence of the whole plasmid. Can you confirm me this to double check?

Reply:
If definitely requires single sequences; I don't remember which other limitations it has off the top of my head. Experiment, and if it doesn't do what you need, then either enhance it or don't use it, as you suggest.